test/test_nestedtensor.py

```diff
@@ -3622,14 +3614,15 @@ def check_forward_backward():
         check_forward_backward()
 
         # Test dispatcher works by calling only mem-effn and math (as they are safe for all devices)
-        with torch.backends.cuda.sdp_kernel(enable_flash=False, enable_mem_efficient=True, enable_math=False):
+        with torch.backends.cuda.sdp_kernel(enable_flash=False, enable_mem_efficient=True, enable_math=True):
             check_forward_backward()
 
-        # Will fail bc unsupported ops
-        # TODO: Add remaining ops, or implement a different math dispatch for jagged
+        # Test math fallback
         with torch.backends.cuda.sdp_kernel(enable_flash=False, enable_mem_efficient=False, enable_math=True):
-            with self.assertRaises(RuntimeError):
-                attn_nt = torch.nn.functional.scaled_dot_product_attention(q_nt_t, k_nt_t, v_nt_t).transpose(1, 2)
+            # Math fallback doesn't work with bfloat16 on CUDA because
+            # "group_gemm_dispatch" not implemented for 'BFloat16'
+            if not (device == "cuda" and dtype == torch.bfloat16):
```

Comment:
Any suggestion on how to bypass this correctly?

Reply:
Skipping here should be fine

Reply:
but the test still fails here (https://github.com/pytorch/pytorch/actions/runs/7335911393/job/19975043647). maybe the device doesn't call "cuda" exactly sometimes?

Reply:
Hmm that is strange, "cuda" should be correct. Maybe try printing out the device?

Reply:
it turns out to be "cuda:0". now it works.